server/catgenome/src/main/java/com/epam/catgenome/manager/heatmap/HeatmapManager.java

```diff
@@ -0,0 +1,390 @@
+        File file = getFile(path);
+        Heatmap heatmap = getHeatmap(heatmapId);
+        heatmap.setCellAnnotationPath(path);
+        checkCellAnnotation(heatmap);
+        heatmapDao.updateCellAnnotation(heatmapId, FileUtils.readFileToByteArray(file), path);
+    }
+
+    @Transactional(propagation = Propagation.REQUIRED)
+    public void deleteHeatmap(final long heatmapId) {
+        Heatmap heatmap = getHeatmap(heatmapId);
+        heatmapDao.deleteHeatmap(heatmapId);
+        biologicalDataItemManager.deleteBiologicalDataItem(heatmap.getBioDataItemId());
+    }
+
+    public Heatmap loadHeatmap(final long heatmapId) {
+        return heatmapDao.loadHeatmap(heatmapId);
+    }
+
+    public List<List<Map<?, String>>> getContent(long heatmapId) throws IOException {
+        Heatmap heatmap = getHeatmap(heatmapId);
+        final String separator = getSeparator(heatmap.getPath());
+        try (InputStream heatmapIS = heatmapDao.loadHeatmapContent(heatmapId);
+                InputStream annotationIS = heatmapDao.loadCellAnnotation(heatmapId)) {
+            return getAnnotatedContent(heatmapIS, annotationIS, heatmap.getCellValueType(), separator);
+        }
+    }
+
+    public Map<String, String> getLabelAnnotation(final long heatmapId) throws IOException {
+        Heatmap heatmap = heatmapDao.loadHeatmap(heatmapId);
+        if (heatmap == null) {
+            return null;
+        } else {
+            final String separator = getSeparator(heatmap.getLabelAnnotationPath());
+            try (InputStream inputStream = heatmapDao.loadLabelAnnotation(heatmapId)) {
+                return inputStream == null ? null : getDataAsMap(inputStream, separator);
+            }
+        }
+    }
+
+    public HeatmapTree getTree(long heatmapId) {
+//        heatmapDao.loadHeatmapRowTree(heatmapId);
+//        heatmapDao.loadHeatmapColumnTree(heatmapId);
+        return new HeatmapTree();
+    }
+
+    private void readHeatmap(Heatmap heatmap) throws IOException {
+        final String path = heatmap.getPath();
+        final String separator = getSeparator(path);
+
+        try (Reader reader = new FileReader(path); BufferedReader bufferedReader = new BufferedReader(reader)) {
+            String line = bufferedReader.readLine();
+            String[] cells = line.split(separator);
+            List<String> columnLabels = new LinkedList<>();
+            for (int i = 1; i < cells.length; i++) {
+                columnLabels.add(cells[i]);
+            }
+            heatmap.setColumnLabels(columnLabels);
+            int columnsNum = cells.length;
+            List<String> rowLabels = new LinkedList<>();
+            Set<String> values = new HashSet<>();
+            Set<Integer> types = new HashSet<>(3);
+            while ((line = bufferedReader.readLine()) != null) {
+                cells = line.split(separator);
+                Assert.isTrue(cells.length == columnsNum, MessagesConstants.ERROR_INCORRECT_FILE_FORMAT);
```

Comment:
Sorry, I missed it at first review. In order to build error message we need to use:
```
messageHelper.getMessage(MessagesConstants.ERROR_INCORRECT_FILE_FORMAT, params...)
```